modules/olm-disconnected-operatorhub-disconnected-mode.adoc

```diff
@@ -0,0 +1,289 @@
+a disconnected mode.
+
+.Prerequisites
+
+* Cluster administrator access to a connected or disconnected {product-title} cluster and its internal registry.
+* Separate workstation with full Internet access.
+* If pushing images to the {product-title} cluster's internal registry, the registry must be exposed with a route.
+* `podman` version 1.5.1+
+
+.Procedure
+
+. **Disable the default OperatorSources.**
++
+Add `disableAllDefaultSources: true` to the spec:
++
+----
+$ oc patch OperatorHub cluster --type json \
+    -p '[{"op": "add", "path": "/spec/disableAllDefaultSources", "value": true}]'
+----
++
+This disables the default OperatorSources that are configured by default during
+a {product-title} installation.
+
+. **Retrieve package lists.**
++
+To get the list of packages that are available for the default OperatorSources,
+run the following `curl` commands from your workstation with Internet access:
++
+----
+$ curl https://quay.io/cnr/api/v1/packages?namespace=redhat-operators > packages.txt
+$ curl https://quay.io/cnr/api/v1/packages?namespace=community-operators >> packages.txt
+$ curl https://quay.io/cnr/api/v1/packages?namespace=certified-operators >> packages.txt
+----
++
+Each package in the new `packages.txt` is an Operator that you could add to your
+disconnected catalog. From this list, you could either pull every Operator or a
+subset that you would like to expose to users.
+
+. **Pull Operator content.**
++
+For a given Operator in the package list, you must pull its content:
++
+----
+$ curl https://quay.io/cnr/api/v1/packages/<namespace>/<operator_name>/<release>
+----
++
+This example uses the etcd Operator:
+
+.. Retrieve the digest:
++
+----
+$ curl https://quay.io/cnr/api/v1/packages/community-operators/etcd/0.0.12
+----
+
+.. From that JSON, take the digest and use it to pull the gzipped archive:
++
+----
+$ curl -XGET https://quay.io/cnr/api/v1/packages/community-operators/etcd/blobs/sha256/8108475ee5e83a0187d6d0a729451ef1ce6d34c44a868a200151c36f3232822b \
+    -o etcd.tar.gz
+----
+
+.. To pull the information out, you must untar the archive into a
+`manifests/<operator_name>/` directory with all the other Operators that you
+want. For example, to untar to an existing directory called `manifests/etcd/`:
++
+----
+$ mkdir -p manifests/etcd/ <1>
+$ tar -xf etcd.tar.gz -C manifests/etcd/
+----
+<1> Create different subdirectories for each extracted archive so that files are not
+overwritten by subsequent extractions for other Operators.
+
+. *Break apart `bundle.yaml` content, if necessary.*
++
+In your new `manifests/<operator_name>` directory, the goal is to get your bundle in the following directory structure:
++
+----
+manifests/
+└── etcd
+    ├── 0.0.12
+    │   ├── clusterserviceversion.yaml
+    │   └── customresourcedefinition.yaml
+    └── package.yaml
+----
++
+If you see files already in this structure, you can skip this step. However, if
```

Comment:
@adellape I think we also need to tell the user to record or pull out the image that is used by the operator versions. We need this for step 8.

Reply:
Added as a new step in https://github.com/openshift/openshift-docs/pull/16892.